### DISCUSSION TITLE
Purge/wipe towers

### DISCUSSION THREAD

**Discussion by Juniper-Rueter** [OrcaSlicer/OrcaSlicer#1336] [Ideas]
Wondering if theres any possibility of adding purge/wipe towers similar to what superslicer has. For MMU units like the ERCF on units without anywhere to purge filament other than the plate having a purge tower seems like a good solution. Superslicer has this implemented already and has automatic volume calculation based on "pigmentation" values for each filament. This pigmentation calculation is nice but is very difficult to calculate compared to how simple calculating purge volume is currently in bambu/orca. If these volumes could be used to automatically calculate the purge tower volumes it would be ideal.

Below is the code I believe superslicer is using for their tower generation, let me know if this isn't correct and I can try and find the correct reference.
https://github.com/supermerill/SuperSlicer/blob/master/src/libslic3r/GCode/WipeTower.cpp
https://github.com/supermerill/SuperSlicer/blob/master/src/libslic3r/GCode/WipeTower.hpp
https://github.com/supermerill/SuperSlicer/blob/master/src/slic3r/GUI/WipeTowerDialog.cpp
https://github.com/supermerill/SuperSlicer/blob/master/src/slic3r/GUI/WipeTowerDialog.hpp

**Comment by Lardram** [comment]
Hi! Any updates on this? I really like this slicer and have being using it with my voron 2.4, I just build an ERCF and don't want to switch to prusa slicer…

**Comment by Kaympe20** [comment]
@brueter Could you find the commits related to the wipe tower so that it could be pulled over?

**Reply by Juniper-Rueter** [reply]
haven't checked much but not that I've seen unfortunately
